apps/console/src/features/applications/components/settings/sign-on-methods/sign-in-method-customization.tsx

```diff
@@ -314,12 +316,28 @@ export const SignInMethodCustomization: FunctionComponent<SignInMethodCustomizat
      * Handles the update button click event.
      */
     const handleUpdateClick = (): void => {
-        eventPublisher.publish("application-sign-in-method-click-update-button");
-        
         if (AdaptiveScriptUtils.isEmptyScript(adaptiveScript)) {
             setAdaptiveScript(AdaptiveScriptUtils.generateScript(steps + 1).join("\n"));
             setIsDefaultScript(true);
         }
+
+        const eventPublisherProperties = {
+            "script-based": !AdaptiveScriptUtils.isDefaultScript(adaptiveScript, steps),
+            "step-based": []
+        };
+
+        updatedSteps.forEach((updatedStep) => {
+            const step = {};
+            updatedStep.options.forEach((element,id) => {
```

Comment:
will there be a scenario where `updatedStep.options` would return undefined? If so we'll gave to handle the check.

```
if (!(Array.isArray(updatedStep?.options) && updatedStep.options.length > 0)) {
    return;
}
```

Reply:
Since this is a forEach loop, code changed to perform the loop inside if condition.

Reply:
Nice 👍